xilem/src/view/async_repeat.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, marker::PhantomData, sync::Arc};
+
+use tokio::task::JoinHandle;
+use xilem_core::{DynMessage, Message, MessageProxy, NoElement, View, ViewId, ViewPathTracker};
+
+use crate::ViewCtx;
+
+pub fn async_repeat<M, F, H, State, Action, Fut>(
```

Comment:
Isn't the name a little bit too specific and misleading?

I think this is more general isn't it, something like `messenger`, `async_messenge`, `proxy` or actually `fork` (as it "forks" the current "thread" with a new closure returning the content of the thread as a future). I mean I'm not sure about the name either^^ but I think `async_repeat` lets the reader/user think, that this is only for repeating stuff (at least that's what I just did, revisiting the code), while this actually starts a different thread with a handle to send messages to change the state while this view is within the view tree.
Where the message could be an enum, being similar to async elm again.

Reply:
I didn't put much thought. Happy to see a PR which renames it.

The main thing to note about this future is that there's no way for it to get a message "back" from the UI thread, so it's quite limited in what it can do.